src/components/PopupWithForm.js

```diff
@@ -2,21 +2,25 @@ import Popup from "./Popup.js";
 
 export default class PopupWithForm extends Popup {
   constructor(selector, handleFormSubmit) {
-    super({ selector });
-    this._popupForm = this._popupElement.querySelector(".modal__form");
+    super(selector);
+    this._popupForm = this._popup.querySelector(".modal__form");
     this._inputList = this._popupForm.querySelectorAll(".modal__input");
     this._handleFormSubmit = handleFormSubmit;
+    this._submitButton = this._popup.querySelector(".modal__button");
+    this._submitButtonText = this._submitButton.textContent;
   }
 
   close() {
     super.close();
     this._popupForm.reset();
   }
 
-  setInputValues(data) {
-    this._inputList.forEach((input) => {
-      input.value = data[input.name];
-    });
+  setButtonText(submit, buttonText = "Saving...") {
+    if (submit) {
+      this._submitButton.textContent = buttonText;
+    } else {
+      this._submitButton.textContent = this._submitButtonText;
+    }
   }
```

Comment:
**Could be improved**

You can make ` setInputValues` in class `PopupWithForm`, which can insert data into inputs:

```js

  setInputValues(data) {
    this._inputList.forEach((input) => {
      // here you insert the `value` by the `name` of the input
      input.value = data[input.name];
    });
  }



```

So, this way you’ll not have to search the inputs of the profile in `index.js`